cli/cmd/common/diff.go

```diff
@@ -0,0 +1,144 @@
+package common
+
+import (
+	"sort"
+	"strings"
+
+	"github.com/google/go-cmp/cmp"
+	"sigs.k8s.io/yaml"
+)
+
+// Diff returns a string representation of the difference between two maps as YAML.
+// The returned string is sorted alphabetically by key. If the maps are identical, the returned string is empty.
+func Diff(a, b map[string]interface{}) (string, error) {
+	if len(a) == 0 && len(b) == 0 {
+		return "", nil
+	}
+
+	buf := new(strings.Builder)
+	err := diffRecursively(a, b, 0, buf)
+
+	return buf.String(), err
+}
+
+// diffRecursively iterates over both maps and writes the differences to the given buffer.
+func diffRecursively(a, b map[string]interface{}, recurseDepth int, buf *strings.Builder) error {
```

Comment:
🚗 drive by comment: you could potentially change the signature of diffRecursively to (a, b map[string]interface{}) (string, err)

Then Diff() can return diffRecursively(a, b, 0) and you wouldn't need to create the stringBuilder inside Diff. IMO it reads more idiomatic.

Reply:
I definitely see the appeal of that. I tried that model too. There is a way to do it with each recursive call creating a string buffer and a way where concatenated strings are just returned. 

The former can be wasteful with generating new buffers. Not really a problem, but easily avoided.

The latter gets messy because the recursion isn't purely linear. It's recursing down a tree structure so you get all these concatenations to the current diff plus where you need to put the values from the next recursion.

I'd be willing to give the second model one more try and see if I can get it clean.

Reply:
hmm. you can still create just the single buffer but create it inside `diffRecursively()`, if that would make the execution in other places simpler.
Also, given we are parsing our helm config which we know does not go very deep in terms of nesting, it should not involve that much recursion so I think we can get away with having multiple buffers.

Reply:
I'll think about it some more and come back to the idea later. I think what is there is ok for now. But I do appreciate this take.

Reply:
I tried to integrate those thoughts here. Let me know what you think! I always appreciate your drive-bys.

Reply:
Just gave it a read! I really like it. Thanks for addressing it :)